runtime/gc_structs/ClassIteratorClassSlots.cpp

```diff
@@ -71,7 +71,14 @@ GC_ClassIteratorClassSlots::nextSlot()
 			return slotPtr;
 		}
 		_state += 1;
-		
+	
+	case classiteratorclassslots_state_default:
```

Comment:
I noticed this is wrong class to modify. This iterator is responsible to iterate `j9class` type slots. However  `_clazzPtr->flattenedClassCache->clazz` actually contain `j9object_t` reference. So a) casts should be done properly and b) scan for this type should be added to `GC_ClassIterator`

Reply:
@dmitripivkine Should I remove this code from here completely and add it at GC_ClassIterator?

Reply:
yes

Reply:
`GC_ClassIteratorClassSlots` returns class slots, `GC_ClassIterator` returns object slots